src/lib/zkstring.ts

```diff
@@ -0,0 +1,148 @@
+import { Bool, Circuit, Field, Poseidon } from '../snarky';
+import { CircuitValue, prop } from './circuit_value';
+
+export class Character extends CircuitValue {
+  @prop value: Field;
+
+  constructor(value: Field) {
+    super();
+
+    if (Character.validateInput(value)) {
```

Comment:
I don't think we should call validateInput in the constructor, the pattern we've been using is to let the constructor be dumb and use .check, see below